mongodb/node-mongodb-native
# feat(NODE-5678): add options parsing support for timeoutMS and defaultTimeoutMS

### Description

#### What is changing?

##### `timeoutMS` option changes

- Add internal `timeoutMS` field to the following option Interfaces
	- `CollectionOptions`
	- `AbstractCursorOptions`
	- `DbOptions`
	- `GridFSBucketReadStreamOptions`
	- `GridFSBucketOptions`
	- `GridFSBucketWriteStreamOptions`
	- `MongoClientOptions`
	- `MongoOptions`
	- `OperationOptions`
- Add internal `timeoutMS` field to `ClientSession`
- Add URI options support for `timeoutMS`

##### `defaultTimeoutMS` option changes

- Add internal `defaultTimeoutMS` field to `ClientSessionOptions`

##### Test changes
- Pull in uri spec tests from DRIVERS-555
- Add `timeoutMS` option support to uri spec runner
- Add `timeoutMS` client option tests
- Add tests confirming expected behaviour of `timeoutMS` option inheritance

##### misc improvements
- Convert 2 test files relevant to this PR to typescript 

##### Is there new documentation needed for these changes?

#### What is the motivation for this change?






### Release Highlight




### Double check the following

- [ ] Ran `npm run check:lint` script
  - Example: `feat(NODE-1234)!: rewriting everything in coffeescript`
